app/views/products/show.html.erb

```diff
@@ -37,12 +37,15 @@
 			<% if @search_terms.any? %>
 		        <%= render partial: 'practices/show/desktop_partials/search_terms', locals: { search_terms: @search_terms } %>
 		  <% end %>
+		  <% if current_user.present? && (current_user&.has_role?(:admin) || current_user == @product.owner || is_user_an_editor_for_innovation(@product, current_user))  %>
+		  	<%= link_to '<i class="fas fa-edit margin-right-1"></i>'.html_safe + 'Edit product', product_editors_path(@product), class: 'usa-button usa-button--secondary margin-top-4 width-full', id: '#show-product-edit-btn' %>
+		  <% end %>
```

Comment:
I think it would be cleaner and more conventional to move this conditional into the controller to be utilized as an instance variable:

```
class ProductsController < ApplicationController
  def show
    @search_terms = @product.categories.get_category_names
    @can_edit_product = current_user.present? && (current_user.has_role?(:admin) || 	 			 	 	 
 	current_user == @product.owner || is_user_an_editor_for_innovation(@product, 
      	current_user))
    render 'products/show'
  end
end
```